### PR TITLE
fix bayesian-methods example

### PR DESCRIPTION
**Pull request by javelinjs**
## Description ##
example/bayesian-methods/sgld.ipynb does not run because some argument position is not correct.

_READY FOR MERGE_

### Essentials ###
- [x] Passed code style checking (`make lint`)
- [x] Changes are complete (i.e. I finished coding on this PR)
- [ ] All changes have test coverage
- [x] For user-facing API changes, API doc string has been updated. For new C++ functions in header files, their functionalities and arguments are well-documented. 
- [x] To my best knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change

### Changes ###
- [x] Fix SGLD optimizer which calls `random.normal` with incorrect arg pos.
- [x] Fix compatible issue with py3

### CODE REVIEW

**Comment by javelinjs** [comment]
@sxjscience But I cannot reproduce the result in paper. SGLD acc for mnist reaches 0.9515 then drops.

**Comment by javelinjs** [comment]
```diff
diff --git a/python/mxnet/optimizer.py b/python/mxnet/optimizer.py
index 2bdebff..1ad8705 100644
--- a/python/mxnet/optimizer.py
+++ b/python/mxnet/optimizer.py
@@ -24,7 +24,7 @@ import logging
 import warnings
 import numpy
 from .base import py_str
-from .ndarray import (NDArray, zeros, clip, sqrt, cast, maximum, abs as NDabs)
+from .ndarray import (NDArray, array, zeros, clip, sqrt, cast, maximum, abs as NDabs)
 from .ndarray import (sgd_update, sgd_mom_update, adam_update, rmsprop_update, rmspropalex_update,
                       mp_sgd_update, mp_sgd_mom_update, square, ftrl_update)
 from .ndarray import _internal
@@ -649,9 +649,11 @@ class SGLD(Optimizer):
         grad = grad * self.rescale_grad
         if self.clip_gradient is not None:
             grad = clip(grad, -self.clip_gradient, self.clip_gradient)
-        weight[:] += - lr/2 * (grad + wd * weight) + normal(0, math.sqrt(lr),
-                                                            shape=weight.shape,
-                                                            ctx=weight.context)
+        import numpy as np
+        rd = np.random.normal(0, math.sqrt(lr), weight.shape)
+        mrd = array(rd, ctx=weight.context)
+        weight[:] += - lr/2 * (grad + wd * weight) + mrd
+        #weight[:] += - lr/2 * (grad + wd * weight) + normal(0, math.sqrt(lr), shape=weight.shape, ctx=weight.context)
```

Then SGLD on MNIST converges to a reasonable acc.

**Comment by piiswrong** [comment]
I thought we needed to fix the c++ random generator?

**Comment by javelinjs** [comment]
@piiswrong Yes. I'm working on it.

**Comment by javelinjs** [comment]
@sxjscience help to verify the changes?

File: example/bayesian-methods/data_loader.py [hunk 1]
```diff
@@ -29,8 +29,7 @@ def load_mnist(training_num=50000):
             'https://github.com/sxjscience/mxnet/raw/master/example/bayesian-methods/mnist.npz'
         )
         print('Downloading data from %s to %s' % (origin, data_path))
-        context = ssl._create_unverified_context()
-        urllib.request.urlretrieve(origin, data_path, context=context)
+        urllib.request.urlretrieve(origin, data_path)
```
Comment (sxjscience) [inline example/bayesian-methods/data_loader.py]: I meet an error such as "Unverified SSL connection" previously when directly using urlretrieve.

Comment (javelinjs) [reply example/bayesian-methods/data_loader.py]: But it seems that there's no `context` arg for `urllib.request.urlretrieve`. I change to a new approach.

**Pull request merged**